protocol/contracts/templegold/TempleGold.sol

```diff
@@ -0,0 +1,340 @@
+
+    /**
+     * @notice Whitelist an address to allow transfer of Temple Gold to or from
+     * @param _contract Contract address to whitelist
+     * @param _whitelist Boolean whitelist state
+     */
+    function authorizeContract(address _contract, bool _whitelist) external override onlyOwner {
+        if (_contract == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        authorized[_contract] = _whitelist;
+        emit ContractAuthorizationSet(_contract, _whitelist);
+    } 
+
+    /**
+     * @notice Set distribution percentages of newly minted Temple Gold
+     * @param _params Distribution parameters
+     */
+    function setDistributionParams(DistributionParams calldata _params) external override onlyOwner {
+        if (_params.staking + _params.gnosis + _params.escrow != DISTRIBUTION_DIVISOR) { revert ITempleGold.InvalidTotalShare(); }
+        distributionParams = _params;
+        emit DistributionParamsSet(_params.staking, _params.escrow, _params.gnosis);
+    }
+
+    /**
+     * @notice Set vesting factor
+     * @param _factor Vesting factor
+     */
+    function setVestingFactor(VestingFactor calldata _factor) external override onlyOwner {
+        if (_factor.numerator == 0 || _factor.denominator == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        if (_factor.numerator > _factor.denominator) { revert CommonEventsAndErrors.InvalidParam(); }
+        vestingFactor = _factor;
+        emit VestingFactorSet(_factor.numerator, _factor.denominator);
+    }
+    
+    /**
+     * @notice Mint new tokens to be distributed. Open to call from any address
+     * Enforces minimum mint amount and uses vesting factor to calculate mint token amount.
+     * Minting is only possible on source chain Arbitrum
+     */
+    function mint() external override onlyArbitrum {
+        VestingFactor memory vestingFactorCache = vestingFactor;
+        DistributionParams storage distributionParamsCache = distributionParams;
+        if (vestingFactorCache.numerator == 0 || distributionParamsCache.escrow == 0) { revert ITempleGold.MissingParameter(); }
+
+        uint256 mintAmount = _getMintAmount(vestingFactorCache);
+        /// @dev no op silently
+        if (!_canDistribute(mintAmount)) { return; }
+
+        lastMintTimestamp = uint32(block.timestamp);
+
+        _distribute(distributionParamsCache, mintAmount);
+    }
+
+    /**
+     * @notice Get vesting factor
+     * @return Vesting factor
+     */
+    function getVestingFactor() external override view returns (VestingFactor memory) {
+        return vestingFactor;
+    }
+
+    /**
+     * @notice Get distribution parameters
+     * @return Distribution parametersr
+     */
+    function getDistributionParameters() external override view returns (DistributionParams memory) {
+        return distributionParams;
+    }
+
+    function canDistribute() external view returns (bool) {
+        VestingFactor memory vestingFactorCache = vestingFactor;
```

Comment:
This is `view` function so no need to cache.
But if this is the code rule, then I understand it.

Reply:
I think it's cheaper to copy into memory in this example because all variables in the struct are used in ` _canDistribute(VestingFactor memory vestingFactorCache) ` and `_getMintAmount(VestingFactor memory vestingFactorCache)`